test/integration_tests/metadata_resource_test.go

```diff
@@ -108,7 +108,7 @@ func TestUpdateWorkflowDefWithTags(t *testing.T) {
 		overwrite    bool
 		expectedTags []model.MetadataTag
 	}{
-		{[]model.MetadataTag{}, false, []model.MetadataTag{tag0}},
+		{[]model.MetadataTag{tag0}, false, []model.MetadataTag{tag0}},
```

Comment:
Why do we need this change? Tags is marked as omitempty

Reply:
because I don't even know how this test completes on CI TBH, given that it never completed locally for me
the left is what we send, the right is what we expect
how can we expect for a tag to be there when we don't send it?

Reply:
digging is how I found out about this
I can only ask @coderabhigupta to chime in

Reply:
@c4lm The first test case was based on running of TestCreateWorkflowDefWithTags before this which will leave tag0 from before. I didn't want to repeat the code and also do clean up only at the end of this test rather than Create. Maybe we can just combine the two tests to avoid confusion.